### PR TITLE
Adds search parameters with and / or logic

### PR DESCRIPTION
**Pull request by rkodev**
Adds search parameters with and / or logic

### CODE REVIEW

**Comment by rkodev** [comment]
This PR has a practical example of a patient search with the equals logic being applied. The example does not have the ability to perform partial searches as the SDK does not yet support the search

File: app/src/main/java/org/smartregister/fhircore/PatientListViewModel.kt [hunk 1]
```diff
@@ -69,11 +70,13 @@ class PatientListViewModel(application: Application, private val fhirEngine: Fhi
...
 
-    private fun getSearchResults(): List<Patient> {
-        val searchResults: List<Patient> = fhirEngine.search()
+    private fun getSearchResults(filterCriterion: FilterCriterion? = null): List<Patient> {
+        val searchBuilder = fhirEngine.search()
             .of(Patient::class.java)
-            //.filter(string(Patient.ADDRESS_CITY, ParamPrefixEnum.EQUAL, "NAIROBI"))
-            .run()
+
+        if(filterCriterion != null) searchBuilder.filter(filterCriterion)
```
Comment (vincent-karuri) [inline app/src/main/java/org/smartregister/fhircore/PatientListViewModel.kt:77]: Not an expert here but can we use`filterCriterion?.let` ?

Comment (rkodev) [reply app/src/main/java/org/smartregister/fhircore/PatientListViewModel.kt]: @vincent-karuri Made a few improvements regarding that

**Review by pld — CHANGES_REQUESTED** [review]
looks good, can you please add test

**Review by vincent-karuri — DISMISSED** [review]

**Comment by rkodev** [comment]
The andorid-fhir SDK has significantly diverged changing the current implementation of the search API. We can merge the PR after refactoring to accommodate search API

**Review by pld — DISMISSED** [review]

**Comment by pld** [comment]
sweat, can you add test for this?

**Comment by rkodev** [comment]
Partially address https://github.com/OpenSRP/fhircore/issues/20

**Review by Rkareko — APPROVED** [review]

**Pull request merged**